source/common/common/matching/url_template_matching.cc

```diff
@@ -0,0 +1,169 @@
+#include "source/common/common/matching/url_template_matching.h"
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "envoy/config/route/v3/route_components.pb.h"
+
+#include "source/common/common/matching/url_template_matching_internal.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "re2/re2.h"
+
+namespace Envoy {
+namespace matching {
+
+using matching::url_template_matching_internal::ParsedUrlPattern;
+
+#ifndef SWIG
+// Silence warnings about missing initializers for members of LazyRE2.
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
+
+inline re2::StringPiece ToStringPiece(absl::string_view text) { return {text.data(), text.size()}; }
+
+bool IsPatternMatch(absl::string_view pattern, absl::string_view capture_regex) {
+  RE2 regex = RE2(ToStringPiece(capture_regex));
```

Comment:
Can we use `Regex::CompiledMatcher` and `Regex::parseRegex` instead?

Reply:
Maybe, I can try it out. What is the advantage?